### ISSUE TITLE
Python process hangs without joining following run_in_thread

### ISSUE THREAD

**Issue by jkaye2012** [andymccurdy/redis-py#576]
Currently, if one uses PubSub.run_in_thread and neglects to stop the thread before the program exits, the Python process will hang on the processing thread. In my testing, this issue is resolved easily by making the worker thread a daemon before starting it; in this way, the processing thread is terminated when its parent dies.

Are there any issues with daemonizing the worker thread in this way?

Commit: 48ae31aabb1497779cadd525a9b6d865d5b8b90a

**Comment by jkaye2012** [comment]
Any word on this?

**Comment by jkaye2012** [comment]
I would really love to not have to fork this project and build my own version of this package.. Can any maintainer comment on this?

**Comment by jackyfkc** [comment]
You could attempt to create a new pull request

**Comment by andymccurdy** [comment]
This was resolved long ago. Sorry for not closing this sooner.
